app/src/main/java/com/cornellappdev/uplift/ui/components/goalsetting/ScrollWheel.kt

```diff
@@ -0,0 +1,242 @@
+package com.cornellappdev.uplift.ui.components.goalsetting
+
+import androidx.compose.foundation.border
+import androidx.compose.foundation.gestures.snapping.rememberSnapFlingBehavior
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Spacer
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.layout.width
+import androidx.compose.foundation.lazy.LazyColumn
+import androidx.compose.foundation.lazy.rememberLazyListState
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableIntStateOf
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.rememberCoroutineScope
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.layout.onGloballyPositioned
+import androidx.compose.ui.layout.positionInParent
+import androidx.compose.ui.platform.LocalDensity
+import androidx.compose.ui.text.TextStyle
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.Dp
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import com.cornellappdev.uplift.util.GRAY01
+import com.cornellappdev.uplift.util.GRAY08
+import com.cornellappdev.uplift.util.montserratFamily
+import kotlinx.coroutines.launch
+import kotlin.math.abs
+
+/**
+ * @param width: the width of the scroll wheel
+ * @param itemHeight: the height of each item in the scroll wheel
+ * @param numberOfDisplayedItems: the number of items displayed at once
+ * @param items: the list of items to display
+ * @param initialItem: the item to select initially
+ * @param itemScaleFact: the scale factor for the selected item
+ * @param textStyle: the text style for the items
+ * @param textColor: the color of the unselected items
+ * @param selectedTextColor: the color of the selected item
+ * @param onItemSelected: the callback for when an item is selected
+ * @return ScrollWheel composable that allows the user to select an item from a list of items
+ */
+@Composable
+fun <T> ScrollWheel(
```

Comment:
DAMN, a generic Composable? This is crazy 🤯 clever stuff with the component design!!